setup.py

```diff
@@ -155,9 +155,14 @@
 ]
 extras_require = {
     ':sys_platform != "win32"': ['terminado>=0.3.3'],
-    'doc': ['Sphinx>=1.1'],
+    'doc': [
+        'Sphinx>=1.1',
```

Comment:
The natural place for the dependencies is `setup.py`. Some of them were already there, I just added the missing ones.

Adding a list of dependencies to `CONTRIBUTING.rst` is a bad idea IMHO.
There should be only a single place for this list, and `setup.py` is the best candidate.
If you want me to add the list to `CONTRIBUTING.rst`, I'll have to remove *all* dependencies from `setup.py`, so either way, `setup.py` will have to be changed.

If you want, I can make a separate PR to add the dependencies to `setup.py`, but it'll have to be merged before this PR.